### PR TITLE
Fix TLS_BadVerify test assertions on macOS

### PR DESCRIPTION
**Pull request by pglass**
### Description

On macOS, we get a different error message for an untrusted TLS certificate which causes a couple of check tests cases to fail. This fixes the assertion so that these tests pass on macOS.

### Testing & Reproduction steps

```
go test ./agent/checks -run TLS_BadVerify
```

### PR Checklist

* [x] updated test coverage
* [ ] external facing docs updated
* [x] not a security concern

### CODE REVIEW

**Review by cthain — APPROVED** [review]
LGTM!

File: agent/checks/check_test.go [hunk 1]
```diff
@@ -836,12 +836,19 @@ func TestCheckHTTP_TLS_BadVerify(t *testing.T) {
...
 		}
-		if !strings.Contains(notif.Output(cid), "certificate signed by unknown authority") {
+		if !isInvalidCertificateError(notif.Output(cid)) {
 			r.Fatalf("should fail with certificate error %v", notif.OutputMap())
 		}
 	})
 }
 
+// isInvalidCertificateError checks the error string for an untrusted certificate error.
+// The specific error message is different on Linux and macOS.
```
Comment (kisunji) [inline agent/checks/check_test.go:846]: Can we add a "this is safe to remove when https://github.com/golang/go/issues/52010 is addressed"

**Comment by kisunji** [comment]
Cause: https://github.com/golang/go/issues/52010

File: agent/checks/check_test.go (hunk 1 above)
Comment (pglass) [reply agent/checks/check_test.go:846]: 👍 https://github.com/hashicorp/consul/pull/15903/commits/f7c028a4c0ee27a6ad79d92ea01413c893c7acb2

**Review by kisunji — APPROVED** [review]

**Pull request merged**